SDCTA/src/pages/IndividualVisualization.tsx

```diff
@@ -1,63 +1,75 @@
 import backIcon from "./backIcon.svg";
 import downloadIcon from "./downloadIcon.svg";
 import "./IndividualVisualization.css";
-
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
 import { TableauEmbed } from "../components/TableauEmbed";
+import { getVisByTitle } from "../api/consumer";
+import { type VisualizationObject } from "../api/data";
+import ErrorBoundary from "../components/ErrorBoundary";
+import { Error404 } from "../components/404";
+const initialState: VisualizationObject = {
+  title: "",
+  analysis: "",
+  link: "",
+  hasCSV: false,
+};
+
+export const IndividualVisualization: React.FC = () => {
+  const [searchParams] = useSearchParams();
+  const [title] = useState(searchParams.get("title"));
+  const [visObj, setVisObj] = useState<VisualizationObject | null>(
+    initialState
+  );
 
-interface IndividualVisualizationProp {
-  url: string;
-  csvlink: string;
-}
+  if (title === "" || title === null) {
```

Comment:
minor: empty string is falsy in JS so this could be condensed to `if (!title) {...}`